kedro/framework/cli/starters.py

```diff
@@ -38,6 +43,81 @@
 }
 _STARTERS_REPO = "git+https://github.com/kedro-org/kedro-starters.git"
 
+
+@define(order=True)
+class KedroStarterSpec:
+    """Specification of custom kedro starter template.
+
+    Args:
+        name: name of the starter (name of the directory)
+        template_path: the path to git repository
+        directory: by default it will look for `name`, optionally override with custom directory.
+        origin: preserved field used by kedro internally to determine where does the starter
+        come from, user do not need to provide this field.
+
+    Example:
+    ::
+        >>> spec = KedroStarterSpec(
+                        name="astro-airflow-iris",
+                        template_path="git+https://github.com/kedro-org/kedro-starters.git",
+                        directory=None
+                                   )
+    """
+
+    name: str
+    template_path: str
+    directory: Optional[str] = None
+    origin: Optional[str] = field(init=False, default=None)
+
+    def __attrs_post_init__(self):
+        if not self.directory:
+            self.directory = self.name  # By default they are the same unless specified.
+
+    def format(self) -> Dict:
```

Comment:
Would doing this as `__str__` be a bit cleaner? So you can do `str(kedro_starter_spec)` directly?

Reply:
Oh, I see this actually returns a `Dict` rather than a `str`. In that case I think `to_dict` would be more conventional method name here.

Reply:
In fact, doesn't `attr` have something built in that would do this? `asdict`?